### PR TITLE
SE-1970 Fix sandbox provisioning failure due to prospectus

### PR DESCRIPTION
**Pull request by pkulkark**
Currently, the sandbox provisioning from master branch fails at [this ansible task](https://github.com/edx/configuration/blob/master/playbooks/roles/nginx/tasks/main.yml#L278-L288) due to prospectus preview variables  `PROSPECTUS_PREVIEW_HTPASSWD_USER` and `PROSPECTUS_PREVIEW_HTPASSWD_PASS`, not being defined by default. 

This PR adds a default value to these variables which can be overridden if needed. It also updates the requirements file path for certs.

**JIRA tickets:** [OSPR-4000](https://openedx.atlassian.net/browse/OSPR-4000)

**Dependencies:** None

**Sandbox URL:**
The following sandbox was deployed using this configuration branch:
- LMS: https://pr22539.sandbox.opencraft.hosting/
- Studio: https://studio.pr22539.sandbox.opencraft.hosting/

**Merge deadline:** ASAP - this is breaking provisioning that was previously working

**Testing Instructions:**
1. Provision a master sandbox using this configuration branch.
2. Verify that it provisions successfully.

**Reviewers**:

- [ ] @xitij2000 

- [ ]  edX reviewer[s] TBD

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @pkulkark! I've created [OSPR-4000](https://openedx.atlassian.net/browse/OSPR-4000) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by nedbat** [comment]
@reillz10 Can you take a look at this?

**Comment by natabene** [comment]
@pkulkark Thank you for your contribution. Please let me know once it is ready for edX review.

File: playbooks/roles/nginx/defaults/main.yml [hunk 1]
```diff
@@ -26,6 +26,8 @@ NGINX_USERS:
 
 # Override these vars for adding user to nginx.htpasswd
 # for prospectus preview basic auth
+PROSPECTUS_PREVIEW_HTPASSWD_USER: 'prospectus'
```
Comment (xitij2000) [inline playbooks/roles/nginx/defaults/main.yml]:
```suggestion
PROSPECTUS_PREVIEW_HTPASSWD_USER: "{{ COMMON_HTPASSWD_USER }}"
```

File: playbooks/roles/nginx/defaults/main.yml [hunk 2]
```diff
@@ -26,6 +26,8 @@ NGINX_USERS:
...
+PROSPECTUS_PREVIEW_HTPASSWD_PASS: 'password'
```
Comment (xitij2000) [inline playbooks/roles/nginx/defaults/main.yml]:
```suggestion
PROSPECTUS_PREVIEW_HTPASSWD_PASS: "{{ COMMON_HTPASSWD_PASS }}"
```

**Review by xitij2000 — APPROVED** [review]
Unless I am missing some context here, I think it would be better to use the common basic auth variables, since they are used in all other instances.

File: playbooks/roles/nginx/defaults/main.yml (hunk 1 above)
Comment (pkulkark) [reply playbooks/roles/nginx/defaults/main.yml]: :+1:

File: playbooks/roles/nginx/defaults/main.yml (hunk 2 above)
Comment (pkulkark) [reply playbooks/roles/nginx/defaults/main.yml]: Makes sense. Changed it to use common basic auth.

**Comment by pkulkark** [comment]
@natabene This is good for edX's review.

**Comment by natabene** [comment]
@pkulkark Thanks for letting me know. @edx/devops This is ready for your review.

**Review by fredsmith — APPROVED** [review]
LGTM

**Comment by openedx-webhooks** [comment]
@pkulkark 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/configuration&entry.867055334=https://github.com/edx/configuration/pull/5592&entry.1484655318=https://github.com/pkulkark&entry.752974735=2019-12-30+06:19&entry.1917517419=2020-01-24+18:24&entry.2133058324=Yes) so we can improve your experience in the future.

**Review by reillz10 — APPROVED** [review]
lgtm :shipit:

**Comment by nedbat** [comment]
@reillz10 @fredsmith Is there some way we can keep prospectus out of the nginx role?  The Open edX installation needs nginx, but should have prospectus at all.

**Comment by fredsmith** [comment]
@nedbat all apps on the edx stack have config built into nginx.  Prospectus doesn't use the edx-django-service role, so it has it's own code.  The short answer is no, we can't pull it from the nginx role.

**Comment by nedbat** [comment]
@fredsmith A tag perhaps? Doesn't it seem unfortunate that installations that don't include prospectus are executing steps that are only about prospectus?

**Pull request merged**